src/runtime/inertia/components/Link.vue

```diff
@@ -100,34 +102,49 @@ const isExternal = computed(() => {
   return typeof props.to === 'string' && hasProtocol(props.to, { acceptRelative: true })
 })
 
+const isLinkActive = computed(() => {
+  if (props.active !== undefined) {
+    return props.active
+  }
+
+  if (!props.to) {
+    return false
+  }
+
+  if (props.exact && page.url === props.to) {
+    return true
+  }
+
+  if (!props.exact && page.url.startsWith(props.to)) {
+    return true
+  }
+
+  return false
+})
+
 const linkClass = computed(() => {
-  const active = isActive.value
+  const active = isLinkActive.value
 
   if (props.raw) {
     return [props.class, active ? props.activeClass : props.inactiveClass]
   }
 
   return ui.value({ class: props.class, active, disabled: props.disabled })
 })
-
-const page = usePage()
-const url = computed(() => props.to ?? props.href ?? '')
-
-const isActive = computed(() => props.active || (!!url.value && (props.exact ? url.value === props.href : page?.url.startsWith(url.value))))
 </script>
 
 <template>
-  <template v-if="!isExternal && !!url">
```

Comment:
I think we need to keep this check to avoid rendering an InertiaLink if no `href`/ `to` props is passed.

Reply:
The `isExternal` computed returns false when there's no `to` prop: https://github.com/nuxt/ui/pull/4134/files#diff-edbff721deed555bd2e0edb16d79d26fd1c4fd7afc0b9aa202d5d159b4ba1134R101